table_filters/data_providers.py

```diff
@@ -15,80 +15,110 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-from omero.sys import ParametersI
-from omero.constants.namespaces import NSBULKANNOTATIONS
+import logging
+
 from omero.model import OriginalFileI
-from omero_parade.utils import get_well_image_ids
+from omero_parade.utils import get_dataset_image_ids, \
+    get_project_image_ids, \
+    get_well_image_ids
+from omeroweb.webgateway.views import _annotations
 
+logger = logging.getLogger(__name__)
 
-def get_dataproviders(request, conn):
-    # Can provide data from any table column
-    plate_id = request.GET.get('plate', None)
-    if plate_id is None:
-        return []
-
-    query_service = conn.getQueryService()
-    params = ParametersI()
-    params.addId(plate_id)
-    query = """select oal from PlateAnnotationLink as oal
-        left outer join fetch oal.child as ch
-        left outer join oal.parent as pa
-        where pa.id=:id
-        and ch.ns='%s'""" % NSBULKANNOTATIONS
-    links = query_service.findAllByQuery(query, params, conn.SERVICE_OPTS)
-    shared_resources = conn.getSharedResources()
+
+def get_table(conn, objtype, objid):
     # Just use the first Table we find
     # TODO: handle multiple tables!?
-    file_id = links[0].child.file.id.val
-    table = shared_resources.openTable(OriginalFileI(file_id),
-                                       conn.SERVICE_OPTS)
+    data = _annotations(None, objtype, objid, conn=conn)['data'][0]
```

Comment:
It's possible to get a ```KeyError``` here if there's no annotations on the Dataset or an ```Index Out Of Bounds Exception``` if there's annotations but none with the ```NSBULKANNOTATIONS``` namespace ('data' is an empty list).

Reply:
In my rush to get this opened I've done a pretty bad job of exceptional condition handling overall. I'll make a full pass as part of fixing up the issues with columns which have spaces in their name.